### DISCUSSION TITLE
Can MOOSE be used to simulate wave equations

### DISCUSSION THREAD

**Discussion by suqingji** [idaholab/moose#21344] [Q&A Modules: General] [answered]
Hi, everyone

I wonder if MOOSE can be used to simulate wave equations.

![image](https://user-images.githubusercontent.com/61961455/174424924-9e285c2b-1897-4a5b-b9de-0480e876c0b7.png)

In the above picture, the first equation is easy to solve, but the second equation makes me confused as I do not how to solve the term of the second time derivatives of a variable.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Yes we do support the computation of double derivatives.
in the code you can retrieve them using the Coupleable interface, and in particular the coupledDotDot routine.

In this recent topic someone worked on a wave equation btw
https://github.com/idaholab/moose/discussions/20925

Guillaume

**Reply by suqingji** [reply]
Thank you for your help
